patches/server/0950-Avoid-respawn-desync-of-player-metadata.patch

```diff
@@ -0,0 +1,86 @@
+-            this.setScore(oldPlayer.getScore());
++            this.getEntityData().markDirtyIfNotDefault(Player.DATA_SCORE_ID); // Paper
+         }
+ 
+         this.enchantmentSeed = oldPlayer.enchantmentSeed;
+         this.enderChestInventory = oldPlayer.enderChestInventory;
+-        this.getEntityData().set(ServerPlayer.DATA_PLAYER_MODE_CUSTOMISATION, (Byte) oldPlayer.getEntityData().get(ServerPlayer.DATA_PLAYER_MODE_CUSTOMISATION));
++        this.getEntityData().markDirtyIfNotDefault(Player.DATA_PLAYER_MODE_CUSTOMISATION); // Paper
+         this.lastSentExp = -1;
+         this.lastSentHealth = -1.0F;
+         this.lastSentFood = -1;
+         // this.recipeBook.copyOverData(entityplayer.recipeBook); // CraftBukkit
+         this.seenCredits = oldPlayer.seenCredits;
+         this.enteredNetherPosition = oldPlayer.enteredNetherPosition;
+-        this.setShoulderEntityLeft(oldPlayer.getShoulderEntityLeft());
+-        this.setShoulderEntityRight(oldPlayer.getShoulderEntityRight());
++        // Paper start
++        this.getEntityData().markDirtyIfNotDefault(Player.DATA_SHOULDER_LEFT);
++        this.getEntityData().markDirtyIfNotDefault(Player.DATA_SHOULDER_RIGHT);
++        // Paper end
+         this.setLastDeathLocation(oldPlayer.getLastDeathLocation());
+     }
+ 
+diff --git a/src/main/java/net/minecraft/world/entity/player/Player.java b/src/main/java/net/minecraft/world/entity/player/Player.java
+index 2e6557a19523d18aecff709de30cb4466b46a9fa..8fbc359d19ae9310378c60905c2223f7075529d4 100644
+--- a/src/main/java/net/minecraft/world/entity/player/Player.java
++++ b/src/main/java/net/minecraft/world/entity/player/Player.java
+@@ -141,8 +141,8 @@ public abstract class Player extends LivingEntity {
+     // CraftBukkit - decompile error
+     private static final Map<Pose, EntityDimensions> POSES = ImmutableMap.<Pose, EntityDimensions>builder().put(Pose.STANDING, Player.STANDING_DIMENSIONS).put(Pose.SLEEPING, Player.SLEEPING_DIMENSIONS).put(Pose.FALL_FLYING, EntityDimensions.scalable(0.6F, 0.6F)).put(Pose.SWIMMING, EntityDimensions.scalable(0.6F, 0.6F)).put(Pose.SPIN_ATTACK, EntityDimensions.scalable(0.6F, 0.6F)).put(Pose.CROUCHING, EntityDimensions.scalable(0.6F, 1.5F)).put(Pose.DYING, EntityDimensions.fixed(0.2F, 0.2F)).build();
+     private static final int FLY_ACHIEVEMENT_SPEED = 25;
+-    private static final EntityDataAccessor<Float> DATA_PLAYER_ABSORPTION_ID = SynchedEntityData.defineId(Player.class, EntityDataSerializers.FLOAT);
+-    private static final EntityDataAccessor<Integer> DATA_SCORE_ID = SynchedEntityData.defineId(Player.class, EntityDataSerializers.INT);
++    public static final EntityDataAccessor<Float> DATA_PLAYER_ABSORPTION_ID = SynchedEntityData.defineId(Player.class, EntityDataSerializers.FLOAT); // Paper
++    public static final EntityDataAccessor<Integer> DATA_SCORE_ID = SynchedEntityData.defineId(Player.class, EntityDataSerializers.INT); // Paper
```

Comment:
We now recommend using ATs again for this because the ATs are per-patch and much nicer to look at (instead of them all being in one BIG HUGE patch).

Reply:
That's should be good